File: cs/src/Ice/Connector.cs
// **********************************************************************
//
// Copyright (c) 2003-2007 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

namespace IceInternal
{

    using System.Net.Sockets;

    public interface Connector
    {
        //
        // Blocking connect. The caller must initialize the new
        // transceiver.
        //
        Transceiver connect(int timeout);

        short type();

        int CompareTo(object obj);
    }

}

File: cpp/src/Ice/Connector.cpp
// **********************************************************************
//
// Copyright (c) 2003-present ZeroC, Inc. All rights reserved.
//
// **********************************************************************

#include <Ice/Connector.h>

using namespace std;
using namespace Ice;
using namespace IceInternal;

Connector::~Connector()
{
    // Out of line to avoid weak vtable
}

IceUtil::Shared* IceInternal::upCast(Connector* p) { return p; }
